### ISSUE TITLE
[Bug]: Section Header label doesn't read from i18n.json?

### ISSUE THREAD

**Issue by Huxpro** [web-infra-dev/rspress#869]
### Version

```sh
System:
    OS: macOS 14.0
    CPU: (12) arm64 Apple M2 Max
    Memory: 14.98 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 123.0.6312.59
    Safari: 17.0
```

### Details

<img width="316" alt="image" src="https://github.com/web-infra-dev/rspress/assets/5563315/2f9d5ce3-504a-4897-9150-acbfae21b213">

<img width="219" alt="image" src="https://github.com/web-infra-dev/rspress/assets/5563315/a64c9dac-1927-4bd6-858c-525c84cb25ee">

Noted that I've tried restarting the dev server, as I've remembered the bug where `i18n.json` won't trigger HMR.

### Reproduce link

N/A

### Reproduce Steps

try the screenshot

**Comment by ssoda01** [comment]
Would you update details about  how it is configured in `_meta.json`

Because the official example itself has a case of configuring section heading tags using `i18n.json`, perhaps this will help 🔗[Overview Page - Rspress](https://rspress.dev/guide/default-theme/overview-page.html#2-configuring-_metajson)

**Comment by 10Derozan** [comment]
I can not reproduce it in main branch.

**Comment by Timeless0911** [comment]
fixed it in #870 https://github.com/web-infra-dev/rspress/pull/870
